src/shared/components/Snackbar/Snackbar.tsx

```diff
@@ -1,33 +1,65 @@
-import React from 'react'
-import Icon from '../Icon'
-import { SnackbarButton, SnackbarParagraph, SnackbarWrapper } from './Snackbar.style'
+import React, { useRef, useState, useEffect } from 'react'
+import { Text } from '@/shared/components'
+import {
+  StyledInnerWrapper,
+  SnackbarButton,
+  SnackbarParagraph,
+  SnackbarHeader,
+  SnackbarWrapper,
+  SnackbarAction,
+  StyledIcon,
+  Submessage,
+} from './Snackbar.style'
 
+export type SnackbarVariant = 'primary' | 'secondary'
+export type IconsType = 'error' | 'success' | 'info'
 export type SnackbarProps = {
-  variant?: 'error' | 'success' | 'info'
+  variant?: SnackbarVariant
+  icon?: IconsType
   message: string
-  buttonText?: string
+  subMessage?: string
+  actionText?: string
+  onActionClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
   onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
 }
 
-const SnackbarComponent: React.ForwardRefRenderFunction<HTMLDivElement, SnackbarProps> = (
-  { variant = 'info', message, buttonText = 'Ok', onClick },
-  ref
-) => {
+const Snackbar: React.FC<SnackbarProps> = ({
+  variant = 'secondary',
+  icon,
+  message,
+  subMessage,
+  actionText,
+  onActionClick,
+  onClick,
+}) => {
+  const ref = useRef<HTMLDivElement>(null)
+  const [height, setHeight] = useState(0)
+
+  useEffect(() => {
+    if (ref.current) {
+      setHeight(ref.current.clientHeight)
+    }
+  }, [])
+
   return (
-    <>
-      <SnackbarWrapper ref={ref}>
-        <SnackbarParagraph>
-          <Icon name={variant} />
-          {message}
+    <SnackbarWrapper variant={variant} snackbarHeight={height}>
+      <StyledInnerWrapper ref={ref} hasSubMessage={!!subMessage}>
+        <SnackbarParagraph variant={variant}>
+          <SnackbarHeader>
+            {icon && <StyledIcon name={icon} />}
+            <Text variant="body2">{message}</Text>
+          </SnackbarHeader>
+          {subMessage && <Submessage>{subMessage}</Submessage>}
+          {actionText && (
+            <SnackbarAction variant="tertiary" onClick={onActionClick}>
+              {actionText}
+            </SnackbarAction>
+          )}
         </SnackbarParagraph>
-        <SnackbarButton onClick={onClick}>{buttonText}</SnackbarButton>
-      </SnackbarWrapper>
-    </>
+        <SnackbarButton onClick={onClick} icon="close" variant="tertiary" size="small" />
```

Comment:
This doesn't seem to properly follow layout in Figma. Is there some good reason to position this button absolutely vs just including it in the layout?

Consider what's happening now:

- The header has fixed height of 32px
- The message has a margin of 10px which isn't multiple of our spacing units
- The message can overlap with the button

If you stick the close button and the title into the container, all of those problems go away. This is also how this is structured in Figma. Design should always use multiples of `4` as margins and paddings. If we use `10px` as margin then this most likely points to a mistake in our implementation